src/EditorFeatures/Core/Implementation/Preview/DifferenceViewerPreview.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Text.Differencing;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Preview
+{
+    internal class DifferenceViewerPreview : IDisposable
+    {
+        private IWpfDifferenceViewer _viewer;
+
+        public DifferenceViewerPreview(IWpfDifferenceViewer viewer)
+        {
+            Contract.ThrowIfNull(viewer);
+            _viewer = viewer;
+        }
+
+        public IWpfDifferenceViewer Viewer => _viewer;
+
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+
+            if (_viewer != null && !_viewer.IsClosed)
+            {
+                _viewer.Close();
+            }
+
+            _viewer = null;
+        }
+
+        ~DifferenceViewerPreview()
+        {
+            // make sure we are not leaking diff viewer
+            // we can't close the view from finalizer thread since it must be same
+            // thread (owner thread) this UI is created.
+            if (_viewer != null)
```

Comment:
How could we hit the finalizer if viewer was null?  We only hit the finalizer if Dispose isn't called.  And if Dispose isn't called then the viewer will be non-null.